File: modules/agent/agents/atlas.go
package agents

import (
	"github.com/woyin/OrangeCoding/modules/ai"
	"github.com/woyin/OrangeCoding/modules/core"
	"github.com/woyin/OrangeCoding/modules/tools"
)

// NewAtlas creates the plan execution agent.
// RoleExecutor, tools: [bash, read_file, write_file, edit_file].
func NewAtlas(provider ai.AiProvider, registry *tools.ToolRegistry, workDir string) *BaseAgent {
	allowedTools := []string{"bash", "read_file", "write_file", "edit_file"}
	return newFilteredAgent(provider, registry, workDir, core.RoleExecutor, allowedTools,
		"You are Atlas, the executor. You execute plans step by step, carrying out each action precisely and reporting progress.")
}

File: packages/agent/src/agents/atlas.ts
/**
 * Atlas - plan execution agent.
 * Ported from modules/agent/agents/atlas.go.
 */

import type { AiProvider } from "@orangecoding/ai";
import { AgentRole } from "@orangecoding/core";
import { ToolRegistry } from "@orangecoding/tools";
import { BaseAgent } from "./base.js";
import { newFilteredAgent } from "./helpers.js";

export function newAtlas(provider: AiProvider, registry: ToolRegistry, workDir: string): BaseAgent {
  return newFilteredAgent(provider, registry, workDir, AgentRole.Executor,
    ["bash", "read_file", "write_file", "edit_file"],
    "You are Atlas, the executor. You execute plans step by step, carrying out each action precisely and reporting progress.");
}
